git_machete/cmd.py

```diff
@@ -1255,27 +1274,29 @@ def squash(cb, fork_commit):
     print(fmt(f"\t`git reset {latest_sha}`"))
 
 
-def diff(branch):
+def diff(branch: Optional[str]) -> None:
+    fp: str = fork_point(branch if branch else current_branch(), use_overrides=True)
     params = \
         (["--stat"] if opt_stat else []) + \
-        [fork_point(branch if branch else current_branch(), use_overrides=True)] + \
+        [fp] + \
         ([f"refs/heads/{branch}"] if branch else []) + \
         ["--"]
     run_git("diff", *params)
 
 
-def log(branch):
+def log(branch: str) -> None:
     run_git("log", "^" + fork_point(branch, use_overrides=True), f"refs/heads/{branch}")
 
 
-def commits_between(earliest_exclusive, latest_inclusive):
+def commits_between(earliest_exclusive: str, latest_inclusive: str) -> List[Hash_ShortHash_Message]:
     # Reverse the list, since `git log` by default returns the commits from the latest to earliest.
     return list(reversed(list(map(
-        lambda x: x.split(":", 2),
+        lambda x: x.split(":", 2),  # type: ignore
```

Comment:
Nit: the error here is:
`git_machete/cmd.py:1294: error: Argument 1 to "map" has incompatible type "Callable[[str], List[str]]"; expected "Callable[[str], Tuple[str, str, str]]"`

So I'd at least convert to lamda's return type to `Tuple` to align it better with the expected type